MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/snapshotter/MapSnapshotter.java

```diff
@@ -401,16 +417,108 @@ public void start(@NonNull SnapshotReadyCallback callback, ErrorHandler errorHan
    *
    * @param styleUrl the style url
    */
-  @Keep
-  public native void setStyleUrl(String styleUrl);
+  public void setStyleUrl(String styleUrl) {
+    fullyLoaded = false;
+    options.withStyleBuilder(null);
+    nativeSetStyleUrl(styleUrl);
+  }
+
+  // TODO: Documentation
+  public void setStyle(Style.Builder builder) {
+    fullyLoaded = false;
+    String uri = builder.getUri() != null ? builder.getUri() : options.builder.getUri();
+    String json = builder.getJson() != null ? builder.getJson() : options.builder.getJson();
+    options.withStyleBuilder(builder);
+
+    if (uri != null) {
+      nativeSetStyleUrl(uri);
+    } else if (json != null) {
+      nativeSetStyleJson(json);
+    }
+  }
```

Comment:
Do you feel there is value to reusing the same instance of Snapshot to generate images with different styles? Imo we can simplify the API to deprecate:

```
  public native void nativeSetStyleUrl(String styleUrl);
  public native void nativeSetStyleJson(String styleJson);
```

and only expose setting the style through the options object.
This will simplify the code a lot and removes the burden to handle too much state.